src/lib/util.js

```diff
@@ -497,13 +531,30 @@ async function processSpecification(spec, processFunction, args, options) {
 
         // Load the page
         // (note HTTP status is 0 when `file://` URLs are loaded)
+        let cacheInfo;
         if (spec.html) {
             await page.setContent(spec.html, loadOptions);
         }
         else {
-            const result = await page.goto(spec.url, loadOptions);
+            let result;
+            try {
+              result = await page.goto(spec.url, loadOptions);
+            } catch (err) {
+              if (reuseExistingData) {
+                return {error: "reuseexistingdata"};
```

Comment:
That's more editorial than anything else, but following on my previous comment not to use an exception, I would prefer if that wasn't reported as an "error" but rather as a status response (there was no error, the result of processing is the equivalent of a 304 in HTTP).

```suggestion
                return {status: "notmodified"};
```

(We could also return a `{status: "ok"}` in cases where processing takes place and `{status: "error"}` when processing fails, but let's focus on the problem at hand)